docs/USAGE-GUIDE.md

```diff
@@ -486,6 +486,30 @@ To use certificate transparency, the requester (usually the client) must call `s
 
 Call `s2n_connection_get_sct_list()` to retrieve the received certificate transparency information. The format of this data is the SignedCertificateTimestampList structure defined in section 3.3 of RFC 6962.
 
+## Session Resumption
+
+TLS handshake sessions are CPU-heavy due to the calculations involved in authenticating a certificate. These calculations can be skipped after the first connection by turning on session resumption. This mechanism stores state from the previous session and uses it to establish the next session, allowing the handshake to skip the costly authentication step while keeping the same cryptographic guarantees. Note that this session state is encrypted by the server, so a server will have to set up an external key in order to do session resumption.
+
+### Stateless Session Resumption
+
+In stateless session resumption the server creates a 'session ticket' from some state after a successful handshake. The server sends this ticket to a client, and the client can send that ticket back to the server during a new connection to skip the authentication step. This mechanism allows servers to avoid storing individual state for each client, and for that reason is the preferred method for resuming a session.
+
+#### A Note on TLS1.3 Session Resumption
+
+Session resumption works differently in versions TLS1.3 and higher. Session ticket messages are now sent immediately after the handshake in "post-handshake" messages, although more tickets can be sent and received anytime after the handshake has completed. Because of this clients need to call s2n_recv after negotiating to receive session ticket messages, as these could arrive anytime post-handshake. Additionally, multiple session tickets may be issued for the same connection.
```

Comment:
```suggestion
In TLS1.2, session ticket messages are sent during the handshake and are automatically received as part of calling s2n_negotiate. They will be available as soon as negotiation is complete.

In TLS1.3, session ticket messages are sent after the handshake as "post-handshake" messages and may not be received as part of calling s2n_negotiate. An s2n-tls server will send tickets immediately after the handshake, so they can be received by calling s2n_recv immediately after the handshake completes. However, other server implementations may send their session tickets later, at any time during the connection. 

Additionally, in TLS1.3, multiple session tickets may be issued for the same connection. <call out the APIs to set the number of tickets>
```

Feels weird to call out "and higher" when we don't know how it'll work in a hypothetical 1.4 :)

I'm also not sure you're clear enough on the difference here. We should be very clear on what TLS1.2 does vs what TLS1.3 does. See if you like any part of my rewrite better.